src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -92,85 +99,992 @@ static ColliderJntSphInit D_80AD2790 = {
+            this->actor.world.pos.y - this->actor.floorHeight < 59.0f) {
+            Vec3f pos = this->actor.world.pos;
+            pos.y = this->actor.floorHeight;
+            func_80033480(globalCtx, &pos, 80.0f, 1, 150, 100, 1);
+            EnPeehat_SpawnDust(globalCtx, this, &pos, 75.0f, 2, 1.05f, 2.0f);
+        }
+        if (this->actor.speedXZ < 0) {
+            this->actor.speedXZ += 0.25f;
+        }
+        this->unk2D4--;
+        if (this->unk2D4 <= 0) {
+            if (this->actor.colChkInfo.health == 0) {
+                EnPeehat_SetStateExplode(this);
+            } else if (this->actor.params < 0) {
+                EnPeehat_Ground_SetStateHover(this);
+                this->riseDelayTimer = 60;
+            } else {
+                EnPeehat_Flying_SetStateFly(this);
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD0FC8.s")
+void EnPeehat_SetStateExplode(EnPeehat* this) {
+    Animation_PlayLoop(&this->skelAnime, &D_060005C4);
+    this->state = PEAHAT_STATE_EXPLODE;
+    this->animTimer = 5;
+    this->unk2E0 = 0.0f;
+    EnPeehat_SetActionFunc(this, &EnPeehat_StateExplode);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD1220.s")
+void EnPeehat_StateExplode(EnPeehat* this, GlobalContext* globalCtx) {
+    EnBom* bomb;
+    s32 pad[2];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD1258.s")
+    if (this->animTimer == 5) {
+        bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.world.pos.x,
+                                   this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0x602, 0);
+        if (bomb != NULL) {
+            bomb->timer = 0;
+        }
+    }
+    this->animTimer--;
+    if (this->animTimer == 0) {
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0x40);
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0x40);
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0x40);
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD1440.s")
+void EnPeehat_Adult_CollisionCheck(EnPeehat* this, GlobalContext* globalCtx) {
+    if ((this->colCylinder.base.acFlags & AC_BOUNCED) || (this->colQuad.base.acFlags & AC_BOUNCED)) {
+        this->colQuad.base.acFlags &= ~AC_BOUNCED;
+        this->colCylinder.base.acFlags &= ~AC_BOUNCED;
+        this->colJntSph.base.acFlags &= ~AC_HIT;
+    } else if (this->colJntSph.base.acFlags & AC_HIT) {
+        this->colJntSph.base.acFlags &= ~AC_HIT;
+        func_8003573C(&this->actor, &this->colJntSph, 1);
+        if (this->actor.colChkInfo.damageEffect == DMG_EFF_NUT) {
+            return;
+        }
+        if (this->actor.colChkInfo.damageEffect == DMG_EFF_LIGHT_ICE_ARROW) {
+            return;
+        }
```

Comment:
```suggestion
        if ((this->actor.colChkInfo.damageEffect != DMG_EFF_NUT) && (this->actor.colChkInfo.damageEffect != DMG_EFF_LIGHT_ICE_ARROW)) {
```
These can be phrased without the early returns, I believe.

Reply:
I feel the same way as above. the early returns makes it clearer sooner that the function does nothing to handle those cases. I'd rather either leave as is, or combine them into a single return.